drivers: sensor: sensirion: shtcx: remove chip property

str4t0m opened the pull request:
Since a long time this driver was using a dts property to choose the specific chip variant instead of properly using compatibles. This PR fixes that:

Removes the dts "chip" property, the compatible should be used instead.

Any existing dts should be converted from something like:

```
test_i2c_shtc3: SHTC3@70 {
	compatible = "sensirion,shtcx";
	reg = <0x70>;
	chip = "shtc3";
	measure-mode = "normal";
	clock-stretching;
};
```

to something like:
```

test_i2c_shtc3: SHTC3@70 {
	compatible = "sensirion,shtc3", "sensirion,shtcx";
	reg = <0x70>;
	measure-mode = "normal";
	clock-stretching;
};
```

Tested with both an shtc3 and and an shtc1 chip.

str4t0m committed:
drivers: sensor: sensirion: shtcx: remove chip property

Removes the dts "chip" property, the compatible should be used instead.

Any existing dts should be converted from something like:

test_i2c_shtc3: shtc3@70 {
	compatible = "sensirion,shtcx";
	reg = <0x70>;
	chip = "shtc3";
	measure-mode = "normal";
	clock-stretching;
};

to something like:

test_i2c_shtc3: shtc3@70 {
	compatible = "sensirion,shtc3", "sensirion,shtcx";
	reg = <0x70>;
	measure-mode = "normal";
	clock-stretching;
};

Signed-off-by: Thomas Stranger <thomas.stranger@outlook.com>

kartben reviewed (changes requested):
Required changes for users of this driver need to be explained in the migration guide

str4t0m committed:
doc: migration-guide: sensor: sensirion: shtcx: chip property

Add release notes for removed chip dts property of the
sensirion shtcx binding.

Signed-off-by: Thomas Stranger <thomas.stranger@outlook.com>
